### ISSUE TITLE
loadBalancerIP is not removed from the node

### ISSUE THREAD

**Issue by zwiy** [kube-vip/kube-vip#225]
**Describe the bug**
I'm running kube-vip only with the service loadbalancing (without cp) on dedicated worker nodes.
I deploy a service with `loadBalancerIP: <my-ip>`.
If I delete the service the IP of the service is getting "re-deployed" on the node and not deleted (see logs).

Why is this released IP of the service re-deployed again?

**To Reproduce**
1. kube-vip deployed as daemonset on dedicated nodes
```
apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: kube-vip-ds
  namespace: kube-system
spec:
  selector:
    matchLabels:
      name: kube-vip-ds
  template:
    metadata:
      labels:
        name: kube-vip-ds
    spec:
      containers:
      - args:
        - manager
        env:
        - name: vip_arp
          value: "true"
        - name: vip_interface
          value: ens192
        - name: port
          value: "6443"
        - name: vip_cidr
          value: "32"
        - name: svc_enable
          value: "true"
        - name: vip_leaderelection
          value: "true"
        - name: vip_leaseduration
          value: "5"
        - name: vip_renewdeadline
          value: "3"
        - name: vip_retryperiod
          value: "1"
        - name: vip_address
        image: plndr/kube-vip:v0.3.5
        imagePullPolicy: Always
        name: kube-vip
        resources: {}
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
            - NET_RAW
            - SYS_TIME
      hostNetwork: true
      nodeSelector:
        infra: "true"
      serviceAccountName: kube-vip
      tolerations:
        - key: infra
          operator: Equal
          value: "true"
          effect: NoSchedule
```
2. Deploy kube-vip-cloud-controller
3. Deploy the following App
```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: source-ip-app-deplyoment
  name: loadbalancer
spec:
  loadBalancerIP: 192.168.1.10
  # externalTrafficPolicy: Local
  ports:
  - port: 80
    protocol: TCP
    targetPort: 8080
  selector:
    app: source-ip-app-deplyoment
  type: LoadBalancer
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: source-ip-app-deplyoment
  name: source-ip-app-deplyoment
spec:
  replicas: 1
  selector:
    matchLabels:
      app: source-ip-app-deplyoment
  strategy: {}
  template:
    metadata:
      labels:
        app: source-ip-app-deplyoment
    spec:
      containers:
      - image: k8s.gcr.io/echoserver:1.4
        name: echoserver
        resources: {}
```

4. Deployed service:
```
$ kubectl get svc
NAME           TYPE           CLUSTER-IP    EXTERNAL-IP     PORT(S)        AGE
loadbalancer   LoadBalancer   10.43.45.40   192.168.1.10   80:30743/TCP   4s
```

5. Delete the service -> the IP should be released from the node. But the IP of the service is still on the node: This are the logs of the kube-vip dameonset pod:
```
time="2021-07-02T12:38:24Z" level=info msg="Broadcasting ARP update for 192.168.1.10 (00:...:88) via ens192"
time="2021-07-02T12:38:27Z" level=info msg="Broadcasting ARP update for 192.168.1.10 (00:...:88) via ens192"
time="2021-07-02T12:38:30Z" level=info msg="Service [loadbalancer] has been addded/modified, it has an assigned external addresses [192.168.1.10]"
time="2021-07-02T12:38:30Z" level=info msg="[LOADBALANCER] Stopping load balancers"
time="2021-07-02T12:38:30Z" level=info msg="[VIP] Releasing the Virtual IP [192.168.1.10]"
time="2021-07-02T12:38:30Z" level=info msg=Stopped
time="2021-07-02T12:38:30Z" level=info msg="Removed [0c686bd7-12a3-4c55-be4d-2f407c75d218] from manager, [0] advertised services remain"
time="2021-07-02T12:38:30Z" level=info msg="Service [loadbalancer] has been deleted"
time="2021-07-02T12:38:30Z" level=warning msg="Re-applying the VIP configuration [192.168.1.10] to the interface [ens192]"
```

**Expected behavior**
The IP of the service should be removed from the node and not beeing "re-applied".

**Environment:**
 - OS/Distro: Ubuntu 20.04
 - Kubernetes Version: v1.20 RKE
 - Kube-vip Version: v0.3.5

**Comment by vinayus** [comment]
Adding to same issue above, here is the output of `ip a`

```
2: ens160: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:50:56:8a:fb:b6 brd ff:ff:ff:ff:ff:ff
    inet 172.16.23.10/21 brd 172.16.23.255 scope global noprefixroute dynamic ens160
       valid_lft 2321sec preferred_lft 2321sec
    inet 172.16.23.149/32 scope global ens160
       valid_lft forever preferred_lft forever
    inet 172.16.23.148/32 scope global ens160
       valid_lft forever preferred_lft forever
```

Here VIP are 23.148 and 23.149, they are retained and are pingable even though service is removed.

**Comment by thebsdbox** [comment]
Found the issue, we'd introduced a race condition with the ARP broadcast and IP address check. I've fixed the logic that will stop the ARP broadcast and re-apply of the address **before** it removes the service. This ensures it's no longer re-applied after the load-balancer is deleted !

**Comment by thebsdbox** [comment]
Fixed in #236

**Comment by vinayus** [comment]
Thank you very much for quick fix. 👍

**Comment by thebsdbox** [comment]
Fixed in new release 0.3.6
